Support multiple static IPs and validate subnets

Honny1 opened the pull request:
This PR introduces comprehensive e2e tests for the multi-IP container feature, which allows users to assign multiple static IP addresses to a single container across single and multi-subnet networks.

Additionally, it refactors network inspection in e2e tests by introducing a reusable `InspectNetwork()` helper function that eliminates duplicate code across the test suite. This helper replaces repetitive manual inspection + JSON unmarshaling patterns with a single, consistent interface.

Fixes: https://issues.redhat.com/browse/RHEL-98277

Related: https://github.com/containers/netavark/pull/1379
Related: https://github.com/containers/container-libs/pull/557

<!--
Thanks for sending a pull request!

For more detailed information, please review our contributing guidelines:
https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests
-->

#### Checklist

Ensure you have completed the following checklist for your pull request to be reviewed:
<!-- Use [x] to mark as done, or click the checkbox after opening PR -->

- [x] Certify you wrote the patch or otherwise have the right to pass it on as an open-source patch by signing all
commits. (`git commit -s`). (If needed, use `git commit -s --amend`).  The author email must match
the sign-off email address. See [CONTRIBUTING.md](https://github.com/containers/podman/blob/main/CONTRIBUTING.md#sign-your-prs)
for more information.
- [x] Referenced issues using `Fixes: #00000` in commit message (if applicable)
- [x] [Tests](https://github.com/containers/podman/tree/main/test#readme) have been added/updated (or no tests are needed)
- [x] [Documentation](https://github.com/containers/podman/blob/main/docs/README.md) has been updated (or no documentation changes are needed)
- [x] All commits pass `make validatepr` (format/lint checks)
- [x] [Release note](https://github.com/kubernetes/community/blob/master/contributors/guide/release-notes.md) entered in the section below (or `None` if no user-facing changes)

#### Does this PR introduce a user-facing change?

<!--
Write `None` if there are no user-facing changes, otherwise enter your release note below.
Include "action required" if users need to take action when upgrading.
-->

```release-note
Support assigning multiple static IP addresses to a single container interface via podman run (e.g., `--network net:ip=...` flag), allowing complex network configurations without requiring the elevated `NET_ADMIN` privilege.
```

Honny1 committed:
Add multi-IP container support

Add tests for multiple static IPs per container across single and multi-subnet networks.
Document --network option for comma-separated IPs and subnet-based IP ordering.

Fixes: https://issues.redhat.com/browse/RHEL-98277

Signed-off-by: Jan Rodák <hony.com@seznam.cz>

Honny1 committed:
Use InspectNetwork helper in e2e tests

Signed-off-by: Jan Rodák <hony.com@seznam.cz>

vikas-goel commented on test/e2e/run_networking_test.go:
```diff
@@ -1237,4 +1219,227 @@ EXPOSE 2004-2005/tcp`, ALPINE)
+		{
+			name:      "multiple static IPs in single subnet",
+			subnets:   []string{"10.86.0.0/16"},
+			staticIPs: []string{"10.86.0.10", "10.86.0.11", "10.86.0.12"},
```
Will be good to add a test case for IP address input in CIDR format.

Honny1 committed:
Vendor: go.podman.io/storage@main, go.podman.io/image/v5@main, go.podman.io/common@main

Signed-off-by: Jan Rodák <hony.com@seznam.cz>

Honny1 committed:
Ignore operational EndpointSettings.IPAddress for static IPs in compat API

Docker treats EndpointSettings.IPAddress as operational/state, not desired create input.
Static address requests should come from IPAMConfig fields. Using both can duplicate
the same IPv4 request and trigger already allocated IPAM errors in compose scenarios.

Ref: moby/moby#46183 (daemon validation), docker/cli#4493 (CLI IPAMConfig), moby/moby#19001 (static IP feature).

Signed-off-by: Jan Rodák <hony.com@seznam.cz>

Luap99 reviewed (approved):
changes LGTM but looks like this needs a rebase
